docs/book/driver-deployment/upgrade.md

```diff
@@ -0,0 +1,30 @@
+<!-- markdownlint-disable MD033 -->
+# vSphere CSI Driver - Upgrade
+
+**Note: To upgrade "vSphere with Tanzu" follow vSphere documentation. Following instructions are applicable to native vanilla Kubernetes Cluster deployed on vSphere.**
+
+Each release of vSphere CSI driver has a different set of RBAC rules and deployment YAML files depending on the version of the vCenter on which Kubernetes Cluster is deployed.
+
+To upgrade from one release to another release of vSphere CSI driver, we recommend to completely delete the driver using the original YAML files, you have used to deploy the driver and then use new YAML files to re-install the vSphere CSI driver.
+
+For example, if you want to upgrade driver from `v2.0.0` release to `v2.0.1` release deployed on the Kubernetes Cluster installed on `vSphere 67u3` release
+
+Uninstall `v2.0.0` driver
+
+```bash
+kubectl delete -f https://raw.githubusercontent.com/kubernetes-sigs/vsphere-csi-driver/master/manifests/v2.0.0/vsphere-67u3/vanilla/deploy/vsphere-csi-controller-deployment.yaml
+kubectl delete -f https://raw.githubusercontent.com/kubernetes-sigs/vsphere-csi-driver/master/manifests/v2.0.0/vsphere-67u3/vanilla/deploy/vsphere-csi-node-ds.yaml
+kubectl delete -f https://raw.githubusercontent.com/kubernetes-sigs/vsphere-csi-driver/master/manifests/v2.0.0/vsphere-67u3/vanilla/rbac/vsphere-csi-controller-rbac.yaml
+```
+
+Wait for vSphere CSI Controller Pod and vSphere CSI Nodes Pods to be completely deleted.
+
+Install `v2.0.1` driver
+
+```bash
+kubectl apply -f https://raw.githubusercontent.com/kubernetes-sigs/vsphere-csi-driver/master/manifests/v2.0.1/vsphere-67u3/vanilla/rbac/vsphere-csi-controller-rbac.yaml
```

Comment:
do we want to point to the manifests in release branch? ToT may change at any point..

Reply:
+1

Reply:
We do not maintain YAML files in each different release branch.  We are maintaining yamls on the master branch here - https://github.com/kubernetes-sigs/vsphere-csi-driver/tree/master/manifests so I suggest using the master branch in our doc, so customers can also get the latest updates and fixes we are making in our deployment YAML files.

Reply:
Hey yeah forgot about to that we maintain YAMLs for each release branch in master. Please ignore my comment.